src/main/java/picard/analysis/StrandSpecificityUtil.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Nils Homer
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.analysis;
+
+
+import htsjdk.samtools.AlignmentBlock;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SamPairUtil;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.OverlapDetector;
+import picard.annotation.Gene;
+import picard.annotation.LocusFunction;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+/** A utility class for inferring strand specificity for RNA-Seq experiments. */
+public class StrandSpecificityUtil {
+
+    public enum StrandSpecificity {NONE, FIRST_READ_TRANSCRIPTION_STRAND, SECOND_READ_TRANSCRIPTION_STRAND}
+
+    /**
+     * This method can be used to infer the strand specificity of an RNA-Seq experiment.
+     *
+     * The strand specificity refers to how the sequencing was performed relative to the strand of transcription.  The
+     * strand specificity of sequencing is determined from mapped reads while the strand of transcription is determined from
+     * a set of genes whose transcription strand and regions are known.
+     *
+     * There are three possible outcomes:
+     * 1. No strand specificity - occurs when there is no correlation between the strands of the reads and the strand of transcription.
+     * 2. First read is on the transcription strand.
+     * 3. First read is on the opposite strand of transcription.
+     * For paired end reads, we assume the second read is on the opposite strand as that of the first (forward/reverse orientation).
```

Comment:
I see below that it is tested.  So perhaps reword to say that you filter rather than assume.